docs: add saml idp rbac related troubleshooting guide in the identity center integration doc

flyinghermit opened the pull request:
issue https://github.com/gravitational/teleport/issues/58206

flyinghermit committed:
add saml idp rbac related troubleshooting guide in ic integration doc

flyinghermit committed:
remove header punctuation

ptgott commented on docs/pages/identity-governance/aws-iam-identity-center/guide.mdx:
```diff
@@ -113,6 +115,21 @@ the same [IdP](../../zero-trust-access/sso/sso.mdx) as your current Identity Cen
 external identity source.
 </Admonition>
 
+<Admonition type="info" title="SAML IdP RBAC">
```
There are a lot of caveats and colored boxes before we get to the main content of the section, so it's difficult for me to tell what this section is actually trying to get me to do.

<img width="719" height="682" alt="image" src="https://github.com/user-attachments/assets/35192728-f363-422f-97fa-f413ac9f05ba" />

Is there a way to structure the section to begin with an overview of what Step 3 entails, then combine the warnings into a single, concise Admonition?

flyinghermit replied:
Admonition removed now, see updates  here  https://github.com/gravitational/teleport/pull/58667#discussion_r2325129721

ptgott commented on docs/pages/identity-governance/aws-iam-identity-center/guide.mdx:
````diff
@@ -321,6 +338,73 @@ automatically removed as well once the plugin is uninstalled.
+}
+```
+
+To resolve RBAC error, first, identify what roles are assigned to the user. 
````
```suggestion
To resolve RBAC errors, first, identify what roles are assigned to the user. 
```
Grammar

r0mant commented on docs/pages/identity-governance/aws-iam-identity-center/guide.mdx:
```diff
@@ -113,6 +115,21 @@ the same [IdP](../../zero-trust-access/sso/sso.mdx) as your current Identity Cen
+In this step, Teleport creates a `saml_idp_service_provider` resource for the 
+AWS IAM Identity Center. 
+
+Starting with Teleport version 18 and role version 8, the SAML IdP service provider 
```
Nit: Link "role version 8" to the appropriate reference section (we had a table highlighting version differences somewhere)?

r0mant commented on docs/pages/identity-governance/aws-iam-identity-center/guide.mdx:
```diff
@@ -113,6 +115,21 @@ the same [IdP](../../zero-trust-access/sso/sso.mdx) as your current Identity Cen
+AWS IAM Identity Center. 
+
+Starting with Teleport version 18 and role version 8, the SAML IdP service provider 
+resources are gated behind `app_labels` label matchers. If you are using Teleport 
```
I think it's pretty safe to say that anyone following this guide anew will be on Teleport 18 and role version v8. Should we move this out of the admonition and just make a part of the base guide?

r0mant commented on docs/pages/identity-governance/aws-iam-identity-center/guide.mdx:
````diff
@@ -113,6 +115,21 @@ the same [IdP](../../zero-trust-access/sso/sso.mdx) as your current Identity Cen
+service provider resource with `app_labels` matching `teleport.dev/origin: aws-identity-center` label.
+```yaml
+  allow:
+    app_labels:
+      'teleport.dev/origin': 'aws-identity-center'
````
This is pretty unintuitive, and definitely should be a part of the guide, not just a note. In addition, I think we should update the [custom role example](https://goteleport.com/docs/identity-governance/aws-iam-identity-center/guide/#creating-custom-identity-center-roles) to use role v8 and set proper app label as well.

r0mant replied:
Also, we don't have to do it now but I wonder if we should have a preset "aws-identity-center" role that includes this app label.

flyinghermit replied:
Agree. I do not like this either. Also suggesting origin label for RBAC does not sit right with me. Long term, I plan to add a new label when the SAML app is created for the identity center and reference that here. Ideally, create a "preset" kind of role and reference that.  @smallinsky and I have been discussing similar ideas to simplify this. 

Good point on updating [custom role example](https://goteleport.com/docs/identity-governance/aws-iam-identity-center/guide/#creating-custom-identity-center-roles) instead. I will do that.

flyinghermit replied:
I have updated this in a new commit: https://github.com/gravitational/teleport/pull/58667/commits/372ad0181c88c7d99129bf86c8439795af42aa47
- The app label reference is added to the pre-requisite and in the [custom role example](https://goteleport.com/docs/identity-governance/aws-iam-identity-center/guide/#creating-custom-identity-center-roles)
- Remove admonition but kept a short description for the step to mention saml service provider is being configured

flyinghermit committed:
Merge branch 'master' into sshah/mention-saml-rbac-in-ic-guide

flyinghermit committed:
review suggestions:
- add app label reference to pre-requisite, ic role example
- remove admonition
- grammar fixes
- keep identity source config description

r0mant commented on docs/pages/identity-governance/aws-iam-identity-center/guide.mdx:
```diff
@@ -257,11 +268,14 @@ sets, for example:
 metadata:
   name: aws-dev-access
 spec:
   allow:
+  allow:
```
Seems we have "allow" twice now.

flyinghermit commented on docs/pages/identity-governance/aws-iam-identity-center/guide.mdx:
```diff
@@ -24,6 +24,11 @@ Teleport to push user and group changes.
 - Teleport Enterprise or Teleport Enterprise Cloud cluster version 17.0 or higher.
 - Administrative access to AWS IAM Identity Center.
+- Teleport preset `editor` role or equivalent to configure plugin resource, integration 
+  resource and SAML IdP service provider resource. If you are using [role version 8](../../reference/access-controls/roles.mdx#role-versions) 
+  to manage access (which is the default role version since Teleport version 18.0), 
+  ensure that your role allows `app_labels` for the AWS IAM Identity Center 
+  resource `'teleport.dev/origin': 'aws-identity-center'`.
```
The `app_labels` needs to match for the admin too, because rbac for all the resources that gets created during plugin installation is checked against user's (admin's) context. That's why I put this in  pre-requisite.

flyinghermit replied:
The imported roles are pinned to role v7 so no need to assign them `app_labels`, until we plan to upgrade that version.

flyinghermit replied:
Added a new "configure access" section https://github.com/gravitational/teleport/pull/58667/commits/bd97e2da7bbf4920304659c0aea9a9df58a08307

flyinghermit committed:
add configure access section

r0mant commented on docs/pages/identity-governance/aws-iam-identity-center/guide.mdx:
````diff
@@ -141,10 +152,91 @@ It may take a few minutes for the initial sync to complete.
+  in Step 3.
+- A permission that allows access to the AWS account and permission set. 
+
+### Configure access to the SAML service provider
+
+First, we will configure an access to the SAML service provider that logs you 
+into the AWS Console.
+
+In the Teleport Web UI, from side navigation menu, select "Zero Trust Access > Roles".
+From the "Roles" UI, click "Create New Role" button. Switch to the YAML editor.
+![Teleport role editor YAML mode](../../../img/igs/entraid/entra-teleport-role-yaml-editor.png)
+
+Copy the role spec shown below and paste it in the role editor to create a new role.
+```yaml
+kind: role
+version: v8
+metadata:
+  name: identity-center-access
+spec:
+  allow:
+    app_labels:
+      'teleport.dev/origin': 'aws-identity-center'
+```
+
+Notice that in the role spec above, the `app_labels` defines a `'teleport.dev/origin': 'aws-identity-center'`
+label. This is the default resource label which Teleport assigns to the SAML service 
+provider resource created for the AWS IAM Identity Center. 
+
+Grant this role to the users who need access to the AWS console. 
````
This role alone wouldn't really give them access to anything would it? They need accounts and permission sets too. I would just keep the section below where you show the full role spec, and explain there that the app label gives access to the SAML provider. Otherwise this example standalone is confusing.

r0mant reviewed (approved):
lgtm with the final comment

flyinghermit committed:
mention access to aws account and perm set again in the saml access paragraph

flyinghermit committed:
Merge branch 'master' into sshah/mention-saml-rbac-in-ic-guide
